hoomd/hpmc/pytest/test_external.py

```diff
@@ -0,0 +1,188 @@
+    sim = simulation_factory(two_particle_snapshot_factory())
+    sim.operations.integrator = mc
+
+    # create harmonic field
+    snapshot = sim.state.get_snapshot()
+    if device.communicator.rank == 0:
+        reference_positions = snapshot.particles.position
+        reference_orientations = snapshot.particles.orientation
+    else:
+        reference_positions = [[0, 0, 0], [0, 0, 0]]
+        reference_orientations = [[1, 0, 0, 0], [1, 0, 0, 0]]
+    lattice = hoomd.hpmc.external.field.Harmonic(
+        reference_positions=reference_positions,
+        reference_orientations=reference_orientations,
+        k_translational=1.0,
+        k_rotational=1.0,
+        symmetries=[[1, 0, 0, 0]])
+    mc.external_potential = lattice
+
+    # create C++ mirror classes and set parameters
+    sim.run(0)
+
+    # make sure objecst are attached
+    sim.operations.remove(mc)
+    assert not mc._attached
+    assert not lattice._attached
+
+
+@pytest.mark.cpu
+def test_harmonic_displacement_energy(device, simulation_factory,
+                                      two_particle_snapshot_factory):
+    """Ensure harmonic displacements result in expected energy."""
+    mc = hoomd.hpmc.integrate.Sphere()
+    mc.shape['A'] = dict(diameter=0, orientable=True)
+
+    # create simulation & attach objects
+    sim = simulation_factory(two_particle_snapshot_factory())
+    sim.operations.integrator = mc
+
+    # create harmonic field
+    k_trans = 1.0
+    snapshot = sim.state.get_snapshot()
+    if device.communicator.rank == 0:
+        reference_positions = snapshot.particles.position
+        reference_orientations = snapshot.particles.orientation
+    else:
+        reference_positions = [[0, 0, 0], [0, 0, 0]]
+        reference_orientations = [[1, 0, 0, 0], [1, 0, 0, 0]]
+    lattice = hoomd.hpmc.external.field.Harmonic(
+        reference_positions=reference_positions,
+        reference_orientations=reference_orientations,
+        k_translational=k_trans,
+        k_rotational=1.0,
+        symmetries=[[1, 0, 0, 0]])
+    mc.external_potential = lattice
+
+    dx = 0.01
+    disp = np.array([dx, 0, 0])
+    lattice.reference_positions = lattice.reference_positions + disp
+
+    # run and check energy
+    sim.run(0)
+    assert np.allclose(lattice.energy,
+                       0.5 * dx**2 * k_trans * sim.state.N_particles)
+
+    # make some moves and make sure the different energies are not zero
+    sim.run(10)
+    assert lattice.energy_rotational != 0.0
+    assert lattice.energy_translational != 0.0
+    assert np.allclose(lattice.energy,
+                       lattice.energy_translational + lattice.energy_rotational)
+
+    # set k_rotational to zero and ensure the rotational energy is zero
+    lattice.k_rotational = 0
+    sim.run(0)
+    assert lattice.energy_rotational == 0.0
+
+    # set k_translational to zero and ensure the translational energy is zero
+    lattice.k_translational = 0
+    sim.run(0)
```

Comment:
This should be unnecessary.

Reply:
Even if the energy property has `requires_run=True`?